### PR TITLE
More fixes needed to test the synchrotron.

### PR DESCRIPTION
**Pull request by NegativeMjark**
It's a bit of a messy PR because I had to move all the .*_synced subs to 10apidoc so that I could use them in some of the earlier tests.

### CODE REVIEW

File: tests/90jira/SYN-627.pl [hunk 1]
```diff
@@ -23,7 +25,9 @@
             my $room_id = $_;
 
             matrix_send_room_text_message( $user, $room_id, body => "$room_id" );
-         } @rooms );
+         } ( head -1, @rooms) );
```
Comment (leonerd) [inline tests/90jira/SYN-627.pl]: If this is the only use of `head` here, you could just use

    $rooms[-1]

to save an entire new CPAN import of a module. `head` is more intended for cases like slicing the result of another function call, or getting more than just the final element.

Comment (NegativeMjark) [inline tests/90jira/SYN-627.pl]: ``head -1, @list`` returns everything *except* the last element?

Comment (NegativeMjark) [inline tests/90jira/SYN-627.pl]: I suppose I could do something like ``@rooms[0..($rooms - 2)]``?

Comment (NegativeMjark) [inline tests/90jira/SYN-627.pl]: Done b36584f

Comment (leonerd) [inline tests/90jira/SYN-627.pl]: Ohyes; head is not tail. Oops. :)
In that case, you could use the  `@rooms[0..@rooms-2]` notation,or just stick with `List::Slice` as it was.

**Comment by leonerd** [comment]
LGTM

**Pull request merged**